

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Glossary &#8212; SWDB 2024 Data Book</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'glossary';</script>
    <link rel="canonical" href="https://allenswdb.github.io/glossary.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Practicalities" href="practicalities/practicalities.html" />
    <link rel="prev" title="Classification Tutorial" href="computational/data-analysis/classification.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/cropped-SummerWorkshop_Header.png" class="logo__image only-light" alt="SWDB 2024 Data Book - Home"/>
    <script>document.write(`<img src="_static/cropped-SummerWorkshop_Header.png" class="logo__image only-dark" alt="SWDB 2024 Data Book - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Data Book for Summer Workshop on the Dynamic Brain
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Experimental Data</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="background/background.html">Background</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="background/Mouse-visual-system.html">Mouse visual system</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/transgenic-tools.html">Transgenic tools</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/experimental-setup.html">Behavioral apparatus</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="background/Neuropixels-electrophysiology.html">Extracellular electrophysiology</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="background/neuropixels-description.html">Neuropixels Probes</a></li>
<li class="toctree-l3"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-quality-metrics.html">Unit Quality Metrics</a></li>
<li class="toctree-l3"><a class="reference internal" href="background/Optotagging.html">Optotagging</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="background/Two-photon-calcium-imaging.html">Calcium imaging</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/Ophys-ephys-comparison.html">Ophys ephys comparison</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/em-connectomics.html">EM Connectomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/CCF.html">Common Coordinate Framework</a></li>
<li class="toctree-l2"><a class="reference internal" href="background/metadata.html">Metadata</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="physiology/physiology.html">Physiology &amp; Behavior</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="physiology/ophys/ophys-overview.html">Optical Physiology</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-background.html">Visual Coding — 2-photon</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-dataset.html">Dataset</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-session-data.html">Getting data from a session</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-stimuli.html">Visual stimuli</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-evoked-response.html">Exercise: Evoked response</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-cross-session-data.html">Cross session data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-coding/vc2p-analysis.html">Analysis files and cell specimens table</a></li>
</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ophys/visual-behavior/VB-Ophys.html">Visual Behavior Ophys</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-behavior/VBO-Dataset.html">Visual Behavior Ophys Dataset</a></li>





<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-behavior/VB-BehaviorSessionData.html">Behavior Session Data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-behavior/VBO-ExperimentData.html">Visual Behavior Ophys Experiment Data</a></li>

<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-behavior/VBO-Tutorial-Compare_trial_types.html">Tutorial Comparing Trial Types</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/visual-behavior/VBO-Tutorial-Downloading_data_and_exploring_the_manifest.html">Tutorial Downloading Data</a></li>
</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ophys/V1DD/V1DD-overview.html">V1 Deep Dive</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/V1DD/V1DD-dataset.html">V1DD Dataset</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/V1DD/V1DD-session-data.html">Accessing V1DD data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/V1DD/V1DD-stimuli.html">Visual stimuli</a></li>

</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ophys/BCI/BCI-overview.html">Brain Computer Interface</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/BCI/BCI-dataset.html">Accessing BCI Data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ophys/BCI/BCI-stimuli.html">BCI Stimuli</a></li>
</ul>
</details></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="physiology/ephys/ephys-overview.html">Electrophysiology</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp.html">Visual Coding — Neuropixels</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-quickstart.html">Quick start guide</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-data-access.html">Accessing Neuropixels Visual Coding Data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-units.html">Units</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-receptive-fields.html">Receptive fields</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-lfp.html">Local field potential (LFP)</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-optotagging.html">Optotagging Analysis</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-session.html">Full example of accessing Visual Coding Neuropixels data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-stimulus.html">Visual Stimuli</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-cheatsheet.html">Cheat sheet for visual coding Neuropixels</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-coding/vcnp-links.html">Other resources</a></li>
</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ephys/visual-behavior/VB-Neuropixels.html">Visual Behavior Neuropixels</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-behavior/VBN-Dataset.html">Visual Behavior Neuropixels Dataset</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-behavior/VBN-SessionData.html">Visual Behavior Neuropixels Session</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-behavior/VBN-Tutorial-Aligning_Neural_Data_to_Stimuli.html">Tutorial Aligning Neural Data to Stimuli</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-behavior/VBN-Tutorial-Aligning_Behavior_Data_to_Task_Events.html">Tutorial Aligning Behavioral Data to Task Events</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/visual-behavior/VBN-Tutorial-Analyzing_LFP_Data.html">Tutorial Analyzing LFP data</a></li>
</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ephys/cell-type-lookup-table/ctlut-background.html">Cell Type Look-up Table</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/cell-type-lookup-table/ctlut-session-data.html">Session data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/cell-type-lookup-table/ctlut-optotagging.html">Optotagging</a></li>





<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/cell-type-lookup-table/ctlut-accessing-data.html">Accessing cell type lookup table data</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/cell-type-lookup-table/ctlut-identifying-tagged-units.html">Identifying tagged neurons</a></li>
</ul>
</details></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="physiology/ephys/np-ultra/npultra-psychedelics.html">Neuropixels Ultra &amp; Psychedelics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/np-ultra/npultra-psychedelics-stimuli.html">NP Ultra &amp; Psychedelics Stimuli and Optotagging protocol</a></li>
<li class="toctree-l4"><a class="reference internal" href="physiology/ephys/np-ultra/npultra-psychedelics-accessing-data.html">Accessing NP Ultra &amp; Psychedelics Data</a></li>
</ul>
</details></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="physiology/stimuli/stimuli.html">Stimuli and Behavioral Tasks</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="physiology/stimuli/passive-visual-stimuli/visual-stimuli-list.html">Visual Stimuli</a></li>
<li class="toctree-l3"><a class="reference internal" href="physiology/stimuli/visual-behavior/VB-Behavior.html">Visual Behavior Task</a></li>
<li class="toctree-l3"><a class="reference internal" href="physiology/stimuli/dynamic-foraging/Dynamic-Foraging.html">Dynamic Foraging</a></li>

</ul>
</details></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="anatomy/anatomy.html">Anatomy</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="anatomy/microns-em/em-background.html">Electron Microscopy</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-structural-data.html">Dataset Basics</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/proofreading.html">Proofreading and Data Quality</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-neuroglancer.html">Neuroglancer</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-dash-apps.html">Dash Apps</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-caveclient-setup.html">Setting up CAVEclient</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-querying-tables.html">Programmatic Access</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-coordinates.html">Coordinate Systems</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="anatomy/microns-em/key-tables.html">CAVE Annotation Tables</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-microns-tables.html">MICrONS Key Tables</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/microns-em/em-v1dd-tables.html">V1DD Key Tables</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-volume-data.html">Imagery and Segmentation</a></li>


<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-skeletons.html">Skeletons</a></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-meshes.html">Meshes</a></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-functional_data.html">MICrONS Functional Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-functional-v1dd.html">V1DD Functional Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="anatomy/microns-em/em-ultrastructure.html">Ultrastructure Resources</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="anatomy/single-cell-morphology/scm-background.html">Single Cell Morphology</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="anatomy/single-cell-morphology/scm-imageprocessing.html">ExaSPIM Image Processing</a></li>
<li class="toctree-l3"><a class="reference internal" href="anatomy/single-cell-morphology/scm-data.html">Single Cell Morphology Data Access</a></li>
</ul>
</details></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Computational Tools</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="computational/data-analysis/data-analysis.html">Data Analysis Tutorials</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="computational/data-analysis/PCA.html">Principal Component Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="computational/data-analysis/regression.html">Regression</a></li>
<li class="toctree-l2"><a class="reference internal" href="computational/data-analysis/GLM.html">Generalized Linear Model (GLM)</a></li>
<li class="toctree-l2"><a class="reference internal" href="computational/data-analysis/classification.html">Classification Tutorial</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Resources</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Glossary</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="practicalities/practicalities.html">Practicalities</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="practicalities/github-code-ocean.html">Creating a Code Ocean capsule synced to GitHub repo</a></li>
<li class="toctree-l2"><a class="reference internal" href="practicalities/code-ocean-collab.html">Collaborating using Code Ocean</a></li>
<li class="toctree-l2"><a class="reference internal" href="practicalities/code-ocean-vs-code.html">Using VS Code in Code Ocean</a></li>
<li class="toctree-l2"><a class="reference internal" href="practicalities/allensdk-pyNWB.html">Allen SDK and pyNWB</a></li>
<li class="toctree-l2"><a class="reference internal" href="practicalities/pyNWB.html">Using Python to Access Neurodata Without Borders-type Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="practicalities/git.html">Git</a></li>

</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="bibliography.html">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributors.html">Contributors and Credits</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/allenswdb/allenswdb.github.io" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/allenswdb/allenswdb.github.io/edit/main/databook/glossary.md" target="_blank"
   class="btn btn-sm btn-source-edit-button dropdown-item"
   title="Suggest edit"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-pencil-alt"></i>
  </span>
<span class="btn__text-container">Suggest edit</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/allenswdb/allenswdb.github.io/issues/new?title=Issue%20on%20page%20%2Fglossary.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/glossary.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>

</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Glossary</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="glossary">
<h1>Glossary<a class="headerlink" href="#glossary" title="Permalink to this heading">#</a></h1>
<dl class="simple glossary">
<dt id="term-Action-potential">Action potential<a class="headerlink" href="#term-Action-potential" title="Permalink to this term">#</a></dt><dd><p>A characteristic signal that appears in excitable cell membranes, which takes the
form of an electric potential difference waveform that propagates down the length
of the cell membrane. In neurons, these indicate neuron activation. See <a class="reference internal" href="#term-Spike"><span class="xref std std-term">Spike</span></a>.</p>
</dd>
<dt id="term-Basket-cell">Basket cell<a class="headerlink" href="#term-Basket-cell" title="Permalink to this term">#</a></dt><dd><p>A type of inhibitory neuron whose synaptic output targets the cell body and
proximal dendrites of excitatory neurons. Many basket cells express the
molecular marker parvalbumin (PV), but not all basket cells are PV+: some
express molecules such as cholecystokinin (CCK). PV basket cells are typically
fast spiking compared to other neurons and are thought to be important for
gain control of network activity and setting the temporal precision of network
activity.</p>
</dd>
<dt id="term-Bipolar-cell">Bipolar cell<a class="headerlink" href="#term-Bipolar-cell" title="Permalink to this term">#</a></dt><dd><p>A subset of VIP cell with a bipolar dendritic arbor. See <a class="reference internal" href="#term-VIP-cell"><span class="xref std std-term">VIP cell</span></a>.</p>
</dd>
<dt id="term-BCI">BCI<a class="headerlink" href="#term-BCI" title="Permalink to this term">#</a></dt><dt id="term-Brain-Computer-Interface">Brain Computer Interface<a class="headerlink" href="#term-Brain-Computer-Interface" title="Permalink to this term">#</a></dt><dd><p>A method of controlling a computer signal through the activity of a neuron. This can be extended to other types of devices (e.g. joysticks or robotic arms). This is also often referred to as “Brain Machine Interface”</p>
</dd>
<dt id="term-CCF">CCF<a class="headerlink" href="#term-CCF" title="Permalink to this term">#</a></dt><dt id="term-Common-Coordinate-Framework">Common Coordinate Framework<a class="headerlink" href="#term-Common-Coordinate-Framework" title="Permalink to this term">#</a></dt><dd><p>The <a class="reference internal" href="background/CCF.html"><span class="doc std std-doc">CCF</span></a> is a a standard 3D reference space for the mouse brain that enables spatial integration of data across modalities.</p>
</dd>
<dt id="term-ChR2">ChR2<a class="headerlink" href="#term-ChR2" title="Permalink to this term">#</a></dt><dt id="term-Channelrhodopsin">Channelrhodopsin<a class="headerlink" href="#term-Channelrhodopsin" title="Permalink to this term">#</a></dt><dd><p>A light-gated ion channel used in the field of optogenetics to control neuronal activity with light.</p>
</dd>
<dt id="term-Container">Container<a class="headerlink" href="#term-Container" title="Permalink to this term">#</a></dt><dd><p><em>There is no consistent use of this term</em>
Most often this refers to the set of recording sessions for a single ophys imaging plane, but can also refer to the set of sessions for an animal.</p>
</dd>
<dt id="term-Cre-line">Cre line<a class="headerlink" href="#term-Cre-line" title="Permalink to this term">#</a></dt><dd><p>The Cre-lox system is a site-specific recombinase technology. Cre-recombinase
is a tyrosine site-specific recombinase that catalyzes the recombination of
DNA between specific sites known as <b>loxP</b> sequences. As used in these
experiments, Cre is used with loxP <a class="reference internal" href="#term-Reporter-line"><span class="xref std std-term">Reporter line</span></a> in order to drive
recombinase of the loxP sites and drive the expression of the reporter. As Cre
is often expressed within a specific gene, this allows the reporter expression
to be restricted to particular subset of cells. For specific lines used, see
the section on <a class="reference internal" href="background/transgenic-tools.html"><span class="doc std std-doc">transgenic tools</span></a>.</p>
</dd>
<dt id="term-Dataset">Dataset<a class="headerlink" href="#term-Dataset" title="Permalink to this term">#</a></dt><dd><p><em>There is no consistent use of this term</em></p>
</dd>
<dt id="term-Driver-line">Driver line<a class="headerlink" href="#term-Driver-line" title="Permalink to this term">#</a></dt><dd><p>A general term for transgenic mouse lines that are engineered to label a
specific cell type or cell population by expressing a specific gene under
the control of the promoter for the cell type or cell population of interest.
A <a class="reference internal" href="#term-Cre-line"><span class="xref std std-term">Cre line</span></a> is a common type of Driver line that allows specific
genes to be expressed when crossed with a <a class="reference internal" href="#term-Reporter-line"><span class="xref std std-term">reporter line</span></a>.
The driver line determines what cell population is targeted, and the
reporter line determines what will be expressed in that specific cell population
(for example, GFP, GCaMP, or Channelrhodopsin).</p>
</dd>
<dt id="term-Ephys">Ephys<a class="headerlink" href="#term-Ephys" title="Permalink to this term">#</a></dt><dd><p>Shorthand for electrophysiology.</p>
</dd>
<dt id="term-Experiment">Experiment<a class="headerlink" href="#term-Experiment" title="Permalink to this term">#</a></dt><dd><p><em>There is no consistent use of this term</em>
It can refer to a stimulus protocol, an entire data collection campaign, or a single session. It is highly ambiguous.</p>
</dd>
<dt id="term-Fast-spiking-neuron">Fast spiking neuron<a class="headerlink" href="#term-Fast-spiking-neuron" title="Permalink to this term">#</a></dt><dt id="term-FSN">FSN<a class="headerlink" href="#term-FSN" title="Permalink to this term">#</a></dt><dt id="term-FSI">FSI<a class="headerlink" href="#term-FSI" title="Permalink to this term">#</a></dt><dd><p>Fast spiking neurons are so called because of their “narrow,” fast action
potentials, specifically as seen in intracellular recordings of a cell in
response to a prolonged step of current. Additionally, with sufficient
current injection fast spiking neurons exhibit fast spike rates, and do
not show frequency adaptation, or slowing of spike rates, over time. In
unlabeled extracellular recordings, units with narrow action potentials are
also referred to as fast spiking neurons. This feature is sometimes used
to putatively label neurons with narrow spikes as particular cell types,
such as <a class="reference internal" href="#term-PV-neuron"><span class="xref std std-term">PV+ neuron</span></a>s, among others.</p>
</dd>
<dt id="term-Fluorophore">Fluorophore<a class="headerlink" href="#term-Fluorophore" title="Permalink to this term">#</a></dt><dd><p>A type of molecule which absorb light and re-emit it at a longer wavelength
in a process called fluorescence. As a result, fluorophores fluoresce only
while exposed to a light source.</p>
</dd>
<dt id="term-GABA">GABA<a class="headerlink" href="#term-GABA" title="Permalink to this term">#</a></dt><dd><p>Gamma-aminobutyric acid (GABA) is the main inhibitory neurotransmitter in the
mammalian brain. In cortex, most GABAergic neurons are local interneurons.</p>
</dd>
<dt id="term-Genetically-encoded-calcium-indicator">Genetically-encoded calcium indicator<a class="headerlink" href="#term-Genetically-encoded-calcium-indicator" title="Permalink to this term">#</a></dt><dt id="term-GECI">GECI<a class="headerlink" href="#term-GECI" title="Permalink to this term">#</a></dt><dd><p>A protein expressed by a cell that will change its fluorescence upon binding
to a Ca<sup>2+</sup> ion. Used to visualize neural activity with fluorescence
microscopy.</p>
</dd>
<dt id="term-GCaMP">GCaMP<a class="headerlink" href="#term-GCaMP" title="Permalink to this term">#</a></dt><dd><p>A family of <a class="reference internal" href="#term-GECI"><span class="xref std std-term">GECI</span></a>. GCaMP was generated by a fusion of the calcium
binding domain of the calmodulin protein with green fluorescent protein (GFP).
In these data we use primarily GCaMP6f as well as some GCaMP6s, fast and slow
variants respectively. These two variants differ in their sensitivity as well
as their kinetics — primarily with regards to their decay. For more see
<span id="id1">Chen <em>et al.</em> [<a class="reference internal" href="bibliography.html#id5" title="Tsai-Wen Chen, Trevor J. Wardill, Yi Sun, Stefan R. Pulver, Sabine L. Renninger, Amy Baohan, Eric R. Schreiter, Rex A. Kerr, Michael B. Orger, Vivek Jayaraman, Loren L. Looger, Karel Svoboda, and Douglas S. Kim. Ultrasensitive fluorescent proteins for imaging neuronal activity. Nature, 499(7458):295–300, July 2013. doi:10.1038/nature12354.">2013</a>]</span>.</p>
</dd>
<dt id="term-GFP">GFP<a class="headerlink" href="#term-GFP" title="Permalink to this term">#</a></dt><dd><p>Green fluorescent protein. Discovered at FHL.</p>
</dd>
<dt id="term-Higher-visual-area">Higher visual area<a class="headerlink" href="#term-Higher-visual-area" title="Permalink to this term">#</a></dt><dt id="term-HVA">HVA<a class="headerlink" href="#term-HVA" title="Permalink to this term">#</a></dt><dd><p>A <strong>higher visual area</strong> is a term for cortical visual areas that receive
input from the primary visual cortex, thus considered to be “higher” in the
visual hierarchy. In primates, higher visual areas include V2, V3, V4, V5, MT,
etc. In the mouse, higher visual areas include: VISl, VIsal, VISpm, VISam,
VISrl among others. For more, see <span id="id2">Glickfeld and Olsen [<a class="reference internal" href="bibliography.html#id13" title="Lindsey L. Glickfeld and Shawn R. Olsen. Higher-Order Areas of the Mouse Visual Cortex. Annual Review of Vision Science, 3:251–273, 2017. doi:10.1146/annurev-vision-102016-061331.">2017</a>]</span>.</p>
</dd>
<dt id="term-Hyperparameter">Hyperparameter<a class="headerlink" href="#term-Hyperparameter" title="Permalink to this term">#</a></dt><dd><p>A free parameter that controls behaviors in machine learning algorithms. These
are distinct from parameters which control behaviors of the models developed by
the algorithms; hyperparameters affect how the algorithm finds the models in
the first place.</p>
</dd>
<dt id="term-Interneuron">Interneuron<a class="headerlink" href="#term-Interneuron" title="Permalink to this term">#</a></dt><dd><p>Also known as a local interneuron: a neuron that has short axons and synapse
exclusively with nearby neurons. In the cortex the term is often used to refer to inhibitory neurons.</p>
</dd>
<dt id="term-Interspike-interval">Interspike interval<a class="headerlink" href="#term-Interspike-interval" title="Permalink to this term">#</a></dt><dt id="term-ISI">ISI<a class="headerlink" href="#term-ISI" title="Permalink to this term">#</a></dt><dd><p>The interspike interval is the the time between two sequential action potentials (spikes) of a neuron. The ISI is used in the quality control of spike-sorting for ephys experiments, assuring that spikes assigned to a unit don’t fall within the refactory period of the neuron (a few milliseconds), indicating that there is contaimination between units.. ISI is also used to characterize firing patterns of neurons.</p>
</dd>
<dt id="term-Intrinsic-signal-imaging">Intrinsic signal imaging<a class="headerlink" href="#term-Intrinsic-signal-imaging" title="Permalink to this term">#</a></dt><dt id="term-0">ISI<a class="headerlink" href="#term-0" title="Permalink to this term">#</a></dt><dd><p>Intrinsic signal imaging, also called ISI, is a method to measure changes in
blood flow associated with neural activity using reflectance of red light on
the brain’s surface, measured using a standard CCD camera. The amount of red
light reflected by the brain tissue increases when oxygenated hemoglobin
perfuses the local region. The timecourse of the ISI signal is slow, and the
magnitude of the reflectance changes are small. As a result, the use of periodic
stimuli can aid in signal detection. A common use of ISI is to map
<a class="reference internal" href="#term-Retinotopy"><span class="xref std std-term">retinotopy</span></a> across the brain surface by moving a slowly drifting bar across
the visual field then measuring the signal in each pixel at the frequency of the
periodic drifting bar. ISI has also been used to identify orientation maps in
species with organized orientation maps like cats and primates, as well as to
map the location of the whisker barrels in somatosensory cortex of the mouse.
For additional papers using ISI to map the organization of the mouse visual
cortex see <span id="id3">Kalatsky and Stryker [<a class="reference internal" href="bibliography.html#id23" title="Valery A. Kalatsky and Michael P. Stryker. New Paradigm for Optical Imaging: Temporally Encoded Maps of Intrinsic Signal. Neuron, 38(4):529–545, May 2003. doi:10.1016/S0896-6273(03)00286-1.">2003</a>]</span> and <span id="id4">Garrett <em>et al.</em> [<a class="reference internal" href="bibliography.html#id12" title="Marina E. Garrett, Ian Nauhaus, James H. Marshel, and Edward M. Callaway. Topography and Areal Organization of Mouse Visual Cortex. Journal of Neuroscience, 34(37):12587–12600, September 2014. doi:10.1523/JNEUROSCI.1124-14.2014.">2014</a>]</span>.</p>
</dd>
<dt id="term-Local-field-potential">Local field potential<a class="headerlink" href="#term-Local-field-potential" title="Permalink to this term">#</a></dt><dt id="term-LFP">LFP<a class="headerlink" href="#term-LFP" title="Permalink to this term">#</a></dt><dd><p>Transient electrical potential generated in nervous tissue by the summed
activity of cells in that tissue. This is typically measured in a lower
temporal-frequency band of less than 250 Hz.</p>
</dd>
<dt id="term-Martinotti-cell">Martinotti cell<a class="headerlink" href="#term-Martinotti-cell" title="Permalink to this term">#</a></dt><dd><p>A Martinotti cell is a particular subtype of SST cell that targets the apical
dendrites of pyramidal cells in layer 1. Martinotti cells are found in layer
2/3 and layer 5.</p>
</dd>
<dt id="term-Minnie-column">Minnie column<a class="headerlink" href="#term-Minnie-column" title="Permalink to this term">#</a></dt><dd><p>A colloquial name for the 100 micron by 100 micron square column of cortex
targeted for the census across layers. This column is a particularly well
proofread collection of cells.</p>
</dd>
<dt id="term-Minnie-dataset">Minnie dataset<a class="headerlink" href="#term-Minnie-dataset" title="Permalink to this term">#</a></dt><dd><p>A colloquial name for the millimeter-scale MICrONs electron microscopy dataset.</p>
</dd>
<dt id="term-NWB">NWB<a class="headerlink" href="#term-NWB" title="Permalink to this term">#</a></dt><dt id="term-Neurodata-Without-Borders">Neurodata Without Borders<a class="headerlink" href="#term-Neurodata-Without-Borders" title="Permalink to this term">#</a></dt><dd><p>A standardized file format for physiology and behavior data. All of our physiology and behavior data is stored in NWB files. The Visual Coding and Visual Behavior data are in NWB files with a hdf backend, while the newer data (V1DD, BCI, Dynamic Foraging, NP Ultra &amp; Psychedelics) have a Zarr backend - which is optimized for cloud access. More info can be found <a class="reference external" href="https://nwb.org/">here</a></p>
</dd>
<dt id="term-Neurogliaform-cell">Neurogliaform cell<a class="headerlink" href="#term-Neurogliaform-cell" title="Permalink to this term">#</a></dt><dd><p>A type of interneuron that makes a diffuse axonal arbor and is thought to release <a class="reference internal" href="#term-GABA"><span class="xref std std-term">GABA</span></a> through both synaptic release and volume transmission, non-selectively inhibiting neurons nearby.</p>
</dd>
<dt id="term-Neuropixels">Neuropixels<a class="headerlink" href="#term-Neuropixels" title="Permalink to this term">#</a></dt><dd><p>A family of devices for obtaining high channel count single unit extracellular
recordings created through a collaborative open science project funded by
Howard Hughes Medical Institute, Gatsby Charitable Trust, the Wellcome Trust,
and the Allen Institute. These devices utilize modern integrated circuit
design to miniaturize aspects of electrophysiology, enabling recordings of
hundred of single units from a single probe with minimal brain damage.
<span id="id5">Jun <em>et al.</em> [<a class="reference internal" href="bibliography.html#id22" title="James J. Jun, Nicholas A. Steinmetz, Joshua H. Siegle, Daniel J. Denman, Marius Bauza, Brian Barbarits, Albert K. Lee, Costas A. Anastassiou, Alexandru Andrei, Çağatay Aydın, Mladen Barbic, Timothy J. Blanche, Vincent Bonin, João Couto, Barundeb Dutta, Sergey L. Gratiy, Diego A. Gutnisky, Michael Häusser, Bill Karsh, Peter Ledochowitsch, Carolina Mora Lopez, Catalin Mitelut, Silke Musa, Michael Okun, Marius Pachitariu, Jan Putzeys, P. Dylan Rich, Cyrille Rossant, Wei-lung Sun, Karel Svoboda, Matteo Carandini, Kenneth D. Harris, Christof Koch, John O'Keefe, and Timothy D. Harris. Fully integrated silicon probes for high-density recording of neural activity. Nature, 551(7679):232–236, November 2017. doi:10.1038/nature24636.">2017</a>]</span> describes these probes; a summary can also be found <a class="reference internal" href="background/neuropixels-description.html"><span class="doc std std-doc">here</span></a>.</p>
</dd>
<dt id="term-Ophys">Ophys<a class="headerlink" href="#term-Ophys" title="Permalink to this term">#</a></dt><dd><p>Shorthand for optical physiology, often in reference to <a class="reference internal" href="#term-Two-photon-calcium-imaging"><span class="xref std std-term">Two-photon calcium imaging</span></a>, but can also include other methods such as fiber photometry.</p>
</dd>
<dt id="term-Optogenetics">Optogenetics<a class="headerlink" href="#term-Optogenetics" title="Permalink to this term">#</a></dt><dd><p>A method for controlling the activity of neurons by expressing light activated
ion channels (using a <a class="reference internal" href="#term-Reporter-line"><span class="xref std std-term">reporter line</span></a> ) in a specific subpopulation of
cells (using a <a class="reference internal" href="#term-Driver-line"><span class="xref std std-term">Driver line</span></a>) to enable temporally precise control of
neural spiking. Spiking can be suppressed or enhanced using different types of
reporters. See <span id="id6">Peron and Svoboda [<a class="reference internal" href="bibliography.html#id37" title="Simon Peron and Karel Svoboda. From cudgel to scalpel: toward precise neural control with optogenetics. Nature Methods, 8(1):30–34, January 2011. doi:10.1038/nmeth.f.325.">2011</a>]</span> for a review on optogenetics as a method.</p>
</dd>
<dt id="term-Optotagging">Optotagging<a class="headerlink" href="#term-Optotagging" title="Permalink to this term">#</a></dt><dd><p>A technique that uses <a class="reference internal" href="#term-Optogenetics"><span class="xref std std-term">optogenetics</span></a> in order to identify neurons that belong to
a specific subpopulation. See: <a class="reference internal" href="background/Optotagging.html"><span class="doc std std-doc">Optotagging</span></a>.</p>
</dd>
<dt id="term-Parvalbumin-positive-interneuron">Parvalbumin-positive interneuron<a class="headerlink" href="#term-Parvalbumin-positive-interneuron" title="Permalink to this term">#</a></dt><dt id="term-PV-neuron">PV+ neuron<a class="headerlink" href="#term-PV-neuron" title="Permalink to this term">#</a></dt><dd><p>Fast spiking neurons, also known as fast spiking interneurons, is a
short-hand for parvalbumin positive GABA-ergic inhibitory interneurons found
in many brain regions that have strong inhibitory effects on neighboring
cells. In experimental preparations where the genetic identity of neurons
can be paired with electrophysiological recordings, PV+ neurons have short
action potentials, occasionally less than 400 µS.</p>
</dd>
<dt id="term-Primary-visual-cortex">Primary visual cortex<a class="headerlink" href="#term-Primary-visual-cortex" title="Permalink to this term">#</a></dt><dt id="term-V1">V1<a class="headerlink" href="#term-V1" title="Permalink to this term">#</a></dt><dt id="term-VISp">VISp<a class="headerlink" href="#term-VISp" title="Permalink to this term">#</a></dt><dd><p>The largest visual area in cortex that receives inputs from the Lateral
geniculate nucleus of thalamus. Often referred to as <b>V1</b> or <b>VISp</b>.</p>
</dd>
<dt id="term-Pyramidal-cell">Pyramidal cell<a class="headerlink" href="#term-Pyramidal-cell" title="Permalink to this term">#</a></dt><dd><p>A type of excitatory neuron with a characteristic cell body shape and apical
dendrite. In visual cortex, pyramidal cells are by far the most common type of
excitatory neuron.</p>
</dd>
<dt id="term-Receptive-field">Receptive field<a class="headerlink" href="#term-Receptive-field" title="Permalink to this term">#</a></dt><dd><p>In a sensory context, the receptive field of a neuron is the region of the stimulus domain in which sensory stimulus needs to lie in order to evoke a response.  For visual cortical cells, for example, the receptive field is the region of visual space in which stimuli can evoke neural responses.  In a computational context, this notion is often generalized multiple dimensions (e.g. space, time, frequency, etc.) and thus equates to the necessary stimulus features that drive neural response (e.g. a localized grating of a specific orientation and frequency).</p>
</dd>
<dt id="term-Regular-Spiking-neuron">Regular Spiking neuron<a class="headerlink" href="#term-Regular-Spiking-neuron" title="Permalink to this term">#</a></dt><dt id="term-RS">RS<a class="headerlink" href="#term-RS" title="Permalink to this term">#</a></dt><dd><p>Neurons that, when injected with a long step of current in the context of
intracellular recordings, show spike frequency adaptation where the rate of
spiking decreases over time. These neurons also have longer (or wider) action
potentials, and lower spike rates even when injected with large currents due
to hyperpolarization after each action potential. These are the most common
type of neurons in the mammalian cortex, and are often associated excitatory
neurons. In extracellular recordings, neurons with longer action potentials
are also sometimes referred to as regular spiking neurons, a feature which is
used to associate these units with specific cell types, such as excitatory
pyramidal neurons among others.</p>
</dd>
<dt id="term-Reporter">Reporter<a class="headerlink" href="#term-Reporter" title="Permalink to this term">#</a></dt><dd><p>An exogenous coding region joined to a promoter sequence or element in an
expression vector that is introduced into cells to provide the means for
measuring the promoter activity
<a class="reference external" href="https://www.promega.com/resources/guides/cell-biology/bioluminescent-reporters/#:~:text=What%20is%20a%20Reporter%20Gene,for%20measuring%20the%20promoter%20activity.">source</a>.</p>
</dd>
<dt id="term-Reporter-line">Reporter line<a class="headerlink" href="#term-Reporter-line" title="Permalink to this term">#</a></dt><dd><p>A <b>reporter line</b> is a transgenic mouse line that is engineered to
express a specific protein that enables monitoring or manipulation of neural
activity (such as GFP, GCaMP, or Channelrhodopsin) under the control of cre or
FLP recombinase, or a tetracycline transactivator system. The gene engineered
into the reporter line will not be expressed unless the protein that controls
reporter gene expression (such as cre or FLP) is present, such as by breeding
a mouse from the reporter line with a mouse from a specific <a class="reference internal" href="#term-Driver-line"><span class="xref std std-term">Driver line</span></a> that expresses the control protein. Injecting a virus that delivers cre
or FLP in a cell type specific manner can also trigger the expression of the
reporter gene.</p>
</dd>
<dt id="term-Retinotopy">Retinotopy<a class="headerlink" href="#term-Retinotopy" title="Permalink to this term">#</a></dt><dt id="term-retinotopic-map">retinotopic map<a class="headerlink" href="#term-retinotopic-map" title="Permalink to this term">#</a></dt><dd><p><b>retinotopy</b> refers to the mapping of visual space on to neural space.
Most visual areas of the brain contain an orderly map of visual space such that
neighboring regions in space are represented by neighboring regions in the brain.
Retinotopic maps are typically measured in terms of altitude (aka vertical retinotopy),
referring to the axis from upper to lower visual field, and and azimuth
(aka horizontal retinotopy), referring to the axis from left to right in space.</p>
</dd>
<dt id="term-ROI">ROI<a class="headerlink" href="#term-ROI" title="Permalink to this term">#</a></dt><dd><p>A <b>region of interest</b> is a general term that describes a subregion of an image.
When used in reference to two photon calcium imaging, an ROI is the mask containing pixels thought to belong to a single neuron.</p>
</dd>
<dt id="term-Saccade">Saccade<a class="headerlink" href="#term-Saccade" title="Permalink to this term">#</a></dt><dd><p>A rapid and ballistic eye movement that shifts the visual field between two fixation points. Mice are not foveal animals, and their eye movements are different from foveal animals (such as humans).</p>
</dd>
<dt id="term-Session">Session<a class="headerlink" href="#term-Session" title="Permalink to this term">#</a></dt><dd><p>A physiological and/or behavioral recording that happens at one time.</p>
</dd>
<dt id="term-Skeleton">Skeleton<a class="headerlink" href="#term-Skeleton" title="Permalink to this term">#</a></dt><dd><p>A linear tree-like structure that defines the shape of a neuron.</p>
</dd>
<dt id="term-Somatostatin-cell">Somatostatin cell<a class="headerlink" href="#term-Somatostatin-cell" title="Permalink to this term">#</a></dt><dd><p>A type of inhibitory interneuron expressing the molecular marker somatostatin (SST, or
sometimes SOM). SST cells tend to target the distal dendrites of excitatory
neurons, and have important roles in regulating the activity of excitatory
neurons.</p>
</dd>
<dt id="term-Spatial-frequency">Spatial frequency<a class="headerlink" href="#term-Spatial-frequency" title="Permalink to this term">#</a></dt><dd><p>How often sinusoidal components of as signal or structure repeat per unit of distance.
When used in reference to drifting gratings, spatial frequency means the distance between the
bars of the grating. Typically measured as cycles per degree.</p>
</dd>
<dt id="term-Targeted-structure">Targeted structure<a class="headerlink" href="#term-Targeted-structure" title="Permalink to this term">#</a></dt><dd><p>The brain region where data was collected from.</p>
</dd>
<dt id="term-Temporal-frequency">Temporal frequency<a class="headerlink" href="#term-Temporal-frequency" title="Permalink to this term">#</a></dt><dd><p>How many complete periods the signal goes through for a given unit of time.
Typically measured in Hertz.</p>
</dd>
<dt id="term-Transgenic-line">Transgenic line<a class="headerlink" href="#term-Transgenic-line" title="Permalink to this term">#</a></dt><dd><p>A mouse line whose genome has been altered by the introduction of one or more
foreign DNA sequences. For these contexts, this typical involves using
<a class="reference internal" href="#term-Cre-line"><span class="xref std std-term">Cre line</span></a>s to drive the expression of a <a class="reference internal" href="#term-Reporter-line"><span class="xref std std-term">Reporter line</span></a> within a
specific subset of cells.</p>
</dd>
<dt id="term-Two-photon-calcium-imaging">Two-photon calcium imaging<a class="headerlink" href="#term-Two-photon-calcium-imaging" title="Permalink to this term">#</a></dt><dd><p>A term for techniques which measure neural activity of neurons by measuring a
fluorescent calcium indicator. These indicators are usually a protein
expressed in a cell, such as <a class="reference internal" href="#term-GCaMP"><span class="xref std std-term">GCaMP</span></a>, often using a specific combination
of <a class="reference internal" href="#term-Driver-line"><span class="xref std std-term">Driver line</span></a> and <a class="reference internal" href="#term-Reporter-line"><span class="xref std std-term">reporter line</span></a>s to express GCaMP in a
specific subset of neurons. Fluorescent dyes can also be used to perform
calcium imaging. At rest a neuron has low levels of calcium, and when the
neuron spikes calcium flows into the neuron and raises the level of calcium,
which binds to the calcium indicator and increases the emitted fluorescence in
a specific wavelength. See <span id="id7">Svoboda and Yasuda [<a class="reference internal" href="bibliography.html#id44" title="Karel Svoboda and Ryohei Yasuda. Principles of Two-Photon Excitation Microscopy and Its Applications to Neuroscience. Neuron, 50(6):823–839, June 2006. doi:10.1016/j.neuron.2006.05.019.">2006</a>]</span> for a review of two-photon
calcium imaging.</p>
</dd>
<dt id="term-Unit">Unit<a class="headerlink" href="#term-Unit" title="Permalink to this term">#</a></dt><dd><p>A putative neuron in extracellular electrophysiology, with varying degrees of
confidence assigned to it. In extracellular electrophysiology, neurons are
referred to as <em>units</em>, because we cannot guarantee that all the spikes
assigned to one unit actually originate from a single cell. Unlike in
two-photon imaging, where you can visualize each neuron throughout the entire
experiment, with electrophysiology we can only “see” a neuron when it fires a
spike. If a neuron moves relative to the probe, or if it’s far away from the
probe, some of its spikes may get mixed together with those from other
neurons. Because of this inherent ambiguity, quality metrics allow you to find
the right units for your analysis. Even highly contaminated units can contain
potentially valuable information about brain states, but certain types of
analysis require more stringent quality thresholds to ensure that all of the
included units are well isolated from their neighbors.</p>
</dd>
<dt id="term-VIP-cell">VIP cell<a class="headerlink" href="#term-VIP-cell" title="Permalink to this term">#</a></dt><dd><p>A type of inhibitory interneuron expressing the molecular marker Vasoactive Intestinal Protein. VIP cells tend to target <a class="reference internal" href="#term-Somatostatin-cell"><span class="xref std std-term">Somatostatin cell</span></a>s rather than excitatory neurons. This role as a “disinhibitory specialist” is thought to be important for context-dependent modulation of cortical activity. Many VIP cells have a characteristic bipolar axon that points along the axis of the cortical column and are thus often called “bipolar cells”.</p>
</dd>
<dt id="term-Waveform">Waveform<a class="headerlink" href="#term-Waveform" title="Permalink to this term">#</a></dt><dt id="term-Spike">Spike<a class="headerlink" href="#term-Spike" title="Permalink to this term">#</a></dt><dd><p>In a system neuroscience setting, this often refers to the voltage over time
measured with an electrode when an individual neuron produces an action
potential.</p>
</dd>
</dl>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="computational/data-analysis/classification.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Classification Tutorial</p>
      </div>
    </a>
    <a class="right-next"
       href="practicalities/practicalities.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Practicalities</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Allen Institute Summer Workshop on the Dynamic Brain
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>